inc/database/class-tmsc-media.php

```diff
@@ -0,0 +1,301 @@
+		}
+		return false;
+	}
+
+	/**
+	 * Set the last updated data hash.
+	 */
+	public function set_last_updated_hash() {
+		if ( ! empty( $this->raw ) ) {
+			$this->update_meta( 'tmsc_last_updated', tmsc_hash_data( $this->raw ) );
+		}
+	}
+
+	/**
+	 * Get title
+	 * @return string
+	 */
+	public function get_title(){
+		$title = ( empty( $this->raw->Title ) ) ? $this->raw->RawTitle : $this->raw->Title;
+		return apply_filters( 'tmsc_set_object_title', $title, $this->raw );
+	}
+
+	/**
+	 * Get date of publication
+	 * @return int unix timestamp
+	 */
+	public function get_pubdate(){
+		return apply_filters( 'tmsc_set_object_pubdate', time(), $this->raw );
+	}
+
+	/**
+	 * Get body
+	 * @return HTML
+	 */
+	public function get_body(){
+		return apply_filters( 'tmsc_set_object_body', '', $this->raw );
+	}
+
+	/**
+	 * Get post slug
+	 * @return string post slug
+	 */
+	public function get_post_name() {
+		return sanitize_title_with_dashes( $this->get_title() );
+	}
+
+	/**
+	 * Get post parent
+	 * @return integer parent post id
+	 */
+	public function get_post_parent() {
+		return ( empty( $this->post_parent ) ) ? 0 : $this->post_parent;
+	}
+
+	/**
+	 * Get post type
+	 * @return string
+	 */
+	public function get_post_type() {
+		return 'attachment';
+	}
+
+	/**
+	 * Get post status
+	 * @return string
+	 */
+	public function get_post_status() {
+		return 'inherit';
+	}
+
+	/**
+	 * Save the final post status
+	 * @return string
+	 */
+	public function save_final_object_status() {
+		$this->object->post_status = $this->get_post_status();
+		$this->update( $this->object );
+	}
+
+	/**
+	 * Get post object
+	 * @return WP_Post
+	 */
+	public function get_post() {
+		return $this->get_object();
+	}
+
+	/**
+	 * Update the post (used in after_save usually)
+	 */
+	public function update() {
+		wp_update_post( $this->object );
+	}
+
+	/**
+	 * Load an existing post if it exists.
+	 */
+	public function load_existing() {
+		$this->object = null;
+		// Check for existing post by legacy ID
+		$legacy_id = $this->get_legacy_id();
+		if ( ! empty( $legacy_id ) ) {
+			$existing_post = tmsc_get_object_by_legacy_id( $legacy_id );
+			if ( ! empty( $existing_post ) ) {
+				$this->object = $existing_post;
+			}
+		}
+	}
+
+	/**
+	 * Save this post
+	 * @return boolean true if successfully saved
+	 */
+	public function save() {
+			/*
+		$this->before_save();
+
+		$this->load_existing();
+		if ( $this->requires_update() ) {
+
+			$this->object = $this->add_attachment();
+
+			if ( empty( $this->object->ID ) ) {
+				return false;
+			}
+
+			// Update queue with post meta.
+			$this->save_meta_data();
+
+			// Save Media Attachments
+			$this->save_media_attachments();
+
+			// Update status.
+			$this->after_save();
+
+			return true;
+		}
+		*/
+		return false;
+	}
+
+	/**
+	 * Save post
+	 * @return WP_Object Object
+	 */
+	public function save_post() {
+
+		// $filename should be the path to a file in the upload directory.
+		$filename = $;
```

Comment:
I am not sure this is valid php code. Please look into this and make sure that filename is assigned to an actual value.